depthcharge/data/parsers.py

```diff
@@ -367,3 +482,32 @@ def _parse_scan_id(scan_str: str | int) -> int:
             pass
 
     raise ValueError("Failed to parse scan number")
+
+
+class ParserFactory:
```

Comment:
I would like the "dispatcher" name here, rather than the factory, since it is not really building the parsers

Reply:
Ah but it does build new parser objects 😉